### DISCUSSION TITLE
Jax docker image finds only CpuDevice

### DISCUSSION THREAD

**Discussion by Loungagna** [ROCm/ROCm#3260] [Q&A]
I follow the Jax Installation route using the docker image. When I run the image and get to the workspace, the command:
`python3 -c 'import jax' 2> /dev/null && echo 'Success' || echo 'Failure'`
results in 
`Success`
So I know my image and jax is working. Now when I run:
`python3 -c 'import jax; print(jax.devices())'`
then I only get:
`[CpuDevice(id=0)]`

whereas my native rocminfo show me properly that I have a GPU:
`Name:                    gfx1102
  Uuid:                    GPU-XX
  Marketing Name:          AMD Radeon RX 7700S
  Vendor Name:             AMD
  Feature:                 KERNEL_DISPATCH
  Profile:                 BASE_PROFILE
  Float Round Mode:        NEAR
  Max Queue Number:        128(0x80)
  Queue Min Size:          64(0x40)
  Queue Max Size:          131072(0x20000)
  Queue Type:              MULTI
  Node:                    1
  Device Type:             GPU`

What am I missing?

The goal is for me to ensure Python3 (to start) uses my AMD GPU when leveraging rocm pytorch and other ML capabilities appropriately.

**Comment by fluidnumericsJoe** [comment]
It seems like the GPU may not be visible within the container.

How are you starting your docker container ? Specifically, are you exposing your AMD GPU to the container with `--device /dev/kfd --device /dev/dri` flags, e.g.
```
docker run -it --device /dev/kfd --device /dev/dri rocm/jax:latest
```

**Comment by Loungagna** [comment]
Yes, I pass the device parameters.

**Comment by fluidnumericsJoe** [comment]
What rocm and amdgpu-dkms version are you running on your host system ? Does it match the version provided in the container (latest is 6.1.1 I believe)

I know that `rocminfo` works on your host system, but what does `rocminfo` or `rocm-smi` show within the container ? Just trying to rule out issues other than a potential issue with the jax installation.

**Comment by zichguan-amd** [comment]
Hi @Loungagna, are you still experiencing this issue? If so, please provide more details on your host ROCm version and output of `rocminfo` inside the container as suggested by https://github.com/ROCm/ROCm/discussions/3260#discussioncomment-9693553.

**Comment by Gg61021277** [comment]
Hi @Loungagna , I have a Makefile with a run target for running a docker container.  Have a look at the options i have used for running the container based on ROCm/pytorch  base image on an Intel machine with RX 7900 XTX GPU. [Makefile here](https://github.com/Gg61021277/rocm-dockerfile/blob/main/Makefile)
